radix-engine/src/system/system_modules/auth/authorization.rs

```diff
@@ -284,45 +286,126 @@ impl Authentication {
     pub fn verify_auth_rule<Y: KernelSubstateApi<SystemLockData> + ClientApi<RuntimeError>>(
         acting_location: ActingLocation,
         auth_zone_id: NodeId,
+        access_rules: &AccessRulesConfig,
         auth_rule: &AccessRuleNode,
+        already_verified_authorities: &mut NonIterMap<String, ()>,
         api: &mut Y,
-    ) -> Result<bool, RuntimeError> {
+    ) -> Result<AuthorizationCheckResult, RuntimeError> {
         match auth_rule {
+            AccessRuleNode::Authority(authority) => {
+                if already_verified_authorities.contains_key(authority) {
+                    return Ok(AuthorizationCheckResult::Authorized);
+                }
+
+                match access_rules.rules.get(authority.as_str()) {
+                    Some(access_rule) => {
+                        // TODO: Add costing for every access rule hop
+                        Self::check_authorization_against_access_rule_internal(
+                            acting_location,
+                            auth_zone_id,
+                            access_rules,
+                            access_rule,
+                            already_verified_authorities,
+                            api,
+                        )
+                    }
+                    None => Ok(AuthorizationCheckResult::Failed(vec![])),
```

Comment:
Should this be a panic?

Reply:
Possibly, this will depend on how authorities + authorization will be implemented